plugins/bufferlines/bufferline.nix

```diff
@@ -125,196 +215,395 @@ in
+        Whether to always show the bufferline.
+      '';
 
-      persistBufferSort = helpers.defaultNullOpts.mkBool true "Whether to make the buffer sort persistent.";
+      auto_toggle_bufferline = defaultNullOpts.mkBool true ''
+        Whether to automatically toggle bufferline.
+      '';
 
-      maxPrefixLength = helpers.defaultNullOpts.mkInt 15 "Maximum prefix length";
+      persist_buffer_sort = defaultNullOpts.mkBool true ''
+        Whether to make the buffer sort persistent.
+      '';
 
-      sortBy = helpers.defaultNullOpts.mkStr "id" "sort by";
+      move_wraps_at_ends = defaultNullOpts.mkBool true ''
+        Whether or not the move command "wraps" at the first or last position.
+      '';
 
-      diagnostics = helpers.defaultNullOpts.mkNullable (
-        with types;
-        either bool (enum [
-          "nvim_lsp"
-          "coc"
-        ])
-      ) false "diagnostics";
+      max_prefix_length = defaultNullOpts.mkInt 15 ''
+        Maximum prefix length used when a buffer is de-duplicated.
+      '';
 
-      diagnosticsIndicator = helpers.defaultNullOpts.mkLuaFn null "Either `null` or a function that returns the diagnostics indicator.";
+      sort_by = defaultNullOpts.mkStr "id" ''
+        How to sort the buffers.
 
-      diagnosticsUpdateInInsert = helpers.defaultNullOpts.mkBool true "Whether diagnostics should update in insert mode";
+        Possible values: `insert_after_current`, `insert_at_end`, `id`, `extension`, `relative_directory`, `directory`, `tabs`.
+        Also accepts a function with a signature `function(buffer_a, buffer_b)` allowing you to compare with custom logic.
+      '';
 
-      offsets = helpers.defaultNullOpts.mkNullable (types.listOf types.attrs) null "offsets";
+      diagnostics =
+        defaultNullOpts.mkEnumFirstDefault
+          [
+            false
+            "nvim_lsp"
+            "coc"
+          ]
+          ''
+            Diagnostics provider.
+
+            Set to `false` to disable.
+          '';
+
+      diagnostics_indicator = defaultNullOpts.mkLuaFn null ''
+        Either `null` or a function that returns the diagnostics indicator.
+      '';
+
+      diagnostics_update_on_event = defaultNullOpts.mkBool true ''
+        Use nvim's diagnostic handler.
+      '';
+
+      offsets = defaultNullOpts.mkNullable (types.listOf types.attrs) null "offsets";
 
       groups = {
-        items = helpers.defaultNullOpts.mkListOf types.attrs [ ] "List of groups.";
+        items = defaultNullOpts.mkListOf types.attrs [ ] "List of groups.";
 
         options = {
-          toggleHiddenOnEnter = helpers.defaultNullOpts.mkBool true "Re-open hidden groups on bufenter.";
+          toggle_hidden_on_enter = defaultNullOpts.mkBool true ''
+            Re-open hidden groups on `BufEnter`.
+          '';
         };
       };
 
       hover = {
-        enabled = mkEnableOption "hover";
+        enabled = lib.mkEnableOption "hover";
```

Comment:
This should almost certainly be `defaultNullOpts.mkBool false "Whether to enable hover."`

`mkEnableOption` sets the default to `false` instead of `null`.